File: BE/rempms-recruitment-service/src/main/java/com/pdev/rempms/recruitmentservice/dto/jobVacancy/JobVacancyResponse.java
package com.pdev.rempms.recruitmentservice.dto.jobVacancy;

import com.pdev.rempms.recruitmentservice.dto.employer.EmployerDTO;
import com.pdev.rempms.recruitmentservice.dto.jobPosition.JobPositionResponseDTO;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class JobVacancyResponse {
    private Integer id;
    private String description;
    private LocalDate closingDate;
    private LocalDate openingDate;
    private boolean govtJob;
    private boolean walksInInterview;
    private boolean partTime;
    private boolean active;
    private String posterUrl;
    private String posterName;
    private Integer employerId;
    private EmployerDTO employer;
    private Integer jobPositionId;
    private String jobVacancyRefNo;
    private JobPositionResponseDTO jobPosition;
}

File: FE/rempms-recruitment-frontend/src/app/model/jobVacancy/JobVacancyResponse.ts
import { Employer } from '../employer/Employer';
import { JobPositionResponse } from '../jobPosition/JobPositionResponse';

export interface JobVacancyResponse {
  id: number;
  description: string;
  openingDate: string;
  closingDate: string;
  govtJob: boolean;
  walksInInterview: boolean;
  partTime: boolean;
  posterUrl: string;
  employerId: number;
  employer: Employer;
  jobPositionId: number;
  jobPosition: JobPositionResponse;
  jobVacancyRefNo: string;
}
